docker/environment.yml

```diff
@@ -3,16 +3,21 @@ channels:
   - conda-forge
 dependencies:
   - python
-  - osmnx=1.5.* # for constructing network graph from OpenStreetMap using Overpass API
+  - osmnx=1.5.*
+  - pandas=2.1.*
   - fpdf2=2.7.*
   - sqlalchemy=1.4.*
-  - nicegui>=1.2.24
-  - cryptography>=41.0.2
+  - nicegui=1.4.13
```

Comment:
A lot of these deps are pinned down to the patch. Can we pin to the minor version instead? Patches should always be compatible across a minor version, and it would allow installers to bring in any important fixes.

Reply:
Nicegui is under rapid development (which is good) and they seem to change syntax regularly (can be a bit frustrating).  I guess I was being explicit about which version I had designed our code to work with.  For other packages, I was following dependabot recommendations.

But having said that, if you think that would be better practice, lets give it a go.  As above, I'll aim to build another version of the image later in the week in case there is further feedback.

Reply:
@gboeing I just had a quick go at rebuilding the image removing the minor version pins; as expected, the web app didn't function with this modification (it doesn't load, due to a bug introduced in a recent minor patch https://github.com/zauberzeug/nicegui/issues/2641 ). Its possible other incompatibilities / syntax changes could have been introduced too by one or other packages (even if they shouldn't). But the current selection of versions works for the Docker image according to my tests.

Given the current 4.4.9 build works, and we'll do a further environment update in a few months with OSMnx 2.0, I'm thinking the following:
- merge this branch into main now, and do a new release
- any reviewed and tested changes from @dapugacheva and @rychennn based off this image can then also be merged into main
- I'll then merge main changes into enhancements (or a branch off that, while I work through them to ensure it remains functional).  
- We arrange for translations of report prose once this is finalised (I believe it should be soon)
- I'll update reporting templates, so those who want to use new templates in English can use Enhancements branch
- Once we have translations we can merge enhancements into main finally, 
- then we can do a 4.5.0 release

then we'll look for a more thorough update of packages/environment and refactor code and do a 5.0 release then, perhaps?

For now, are you okay with the merging of this branch as is?